contracts/ERC721A.sol

```diff
@@ -633,6 +648,7 @@ contract ERC721A is IERC721A {
             _packedOwnerships[tokenId] =
                 _addressToUint256(from) |
                 (block.timestamp << BITPOS_START_TIMESTAMP) |
+                (_extraData(from, address(0), _unpackExtraData(prevOwnershipPacked)) << BITPOS_EXTRA_DATA) |
                 BITMASK_BURNED |
                 BITMASK_NEXT_INITIALIZED;
```

Comment:
One of my new tests for the behavior of  `_extraData` failed for burns on my machine. I was able to track it back to the `_addressToUint256(from)` part in the above code. Replacing it with `(_addressToUint256(from) & ((1 << 160) - 1)) |`, so zeroing the leading 96bits solved the issue. I have no idea why `_addressToUint256` seems to produce dirty `uint256`s though. @Vectorized

Reply:
Ok, i figured out.

This is because `extraData` is not at the last 24 bits of the uint256.

For efficiency reasons, and for backwards storage compatibility reasons (we have a upgradeable repo), 
we will need to use the following layout:
```solidity
    // Bits Layout:
    // - [0..159]   `addr`
    // - [160..223] `startTimestamp`
    // - [224]      `burned`
    // - [225]      `nextInitialized`
    // - [232..255] `extraData`
```

Reply:
Hmm... strange.

Anyway, i also encountered the same issue. I've edited some stuff to solve it.

Reply:
It looks like the compiler didn't clean the bits.

I have added some changes to ensure that.